packages/slice-machine/src/hooks/useServerState.tsx

```diff
@@ -2,20 +2,23 @@ import { useEffect } from "react";
 
 import ServerState from "@lib/models/server/ServerState";
 import useSliceMachineActions from "@src/modules/useSliceMachineActions";
+import useSwr from "swr";
+import axios, { AxiosResponse } from "axios";
 
-const useServerState = (serverState: ServerState) => {
-  const { getState } = useSliceMachineActions();
+const fetcher = (url: string): Promise<ServerState> =>
+  axios.get(url).then((res: AxiosResponse<ServerState>) => res.data);
```

Comment:
`axios.get<ServerState>` and the res will be inferred for you :)

Reply:
Typescript can't inferred the `then` type if I use the generics, don't know why